### PR TITLE
Refresh accessToken when JWT access token has expired

### PR DESCRIPTION
**Pull request by mohammadranjbarz**
related #1007

### CODE REVIEW

File: packages/frontend/src/utils/axios.ts [hunk 1]
```diff
@@ -9,10 +10,45 @@ const isJsonBlob = (data): data is Blob =>
...
+  if (
+    refreshToken &&
+    err?.response?.status === 401 &&
+
+    // 1092, 1107 are the error codes for invalid jwt token that defined in backend
+    (err?.response?.data?.code === 1092 || err?.response?.data?.code === 1107)
+  ) {
+    // delete the old token from localStorage to prevent infinite loop
+    delete recoilPersist!['ActiveTokenSet'];
+    localStorage.setItem(JSON.stringify(recoilPersist), 'recoil-persist');
```
Comment (mohammadranjbarz) [inline packages/frontend/src/utils/axios.ts:42]: @kristoferlund 
It seems we just can use `recoil` inside components so I had to manually put/get to local storage

File: packages/frontend/src/utils/axios.ts [hunk 2]
```diff
@@ -9,10 +10,48 @@ const isJsonBlob = (data): data is Blob =>
...
+  if (recoilPersist) {
+    refreshToken = JSON.parse(recoilPersist)?.ActiveTokenSet?.refreshToken;
+  }
+
+  if (
+    refreshToken &&
+    recoilPersist &&
+    err?.response?.status === 401 &&
+    // 1092, 1107 are the error codes for invalid jwt token that defined in backend
+    (err?.response?.data?.code === 1092 || err?.response?.data?.code === 1107)
```
Comment (kristoferlund) [inline packages/frontend/src/utils/axios.ts:38]: 1092 means unauthenticated user trying to access authenticated resource. This should not trigger a refresh.

File: packages/frontend/src/utils/axios.ts [hunk 3]
```diff
@@ -9,10 +10,48 @@ const isJsonBlob = (data): data is Blob =>
...
-): AxiosError => {
+): Promise<
+  AxiosError<{
+    code: number;
+    message: string;
+    statusCode: number;
+  }>
+> => {
+  let refreshToken;
+  const recoilPersist = localStorage.getItem('recoil-persist');
```
Comment (kristoferlund) [inline packages/frontend/src/utils/axios.ts:28]: Use `getRecoil(ActiveTokenSet)`, see `auth.ts` for example.

**Review by kristoferlund — CHANGES_REQUESTED** [review]
Sorry for slow review @mohammadranjbarz! 

I'd like to see two small changes, other than that all good!

Or btw. Have you tried how the user experience is when token expires and gets refreshed? Since the original request fails, the UI is left in an unclear state. Should we force a UI reload to make sure all looks good?

**Comment by kristoferlund** [comment]
@mohammadranjbarz I finished this as I want to get it out to communities soon.

**Review by kristoferlund — APPROVED** [review]

**Comment by mohammadranjbarz** [comment]
> @mohammadranjbarz I finished this as I want to get it out to communities soon.

Thanks @kristoferlund , sorry I was busy last week , I couldn't spend time on this

**Pull request merged**